docs, ci, test/l4lb: use latest cilium-cli release according to stable.txt

tklauser opened the pull request:
This will make sure users download the latest release marked as stable
by developers in stable.txt on the master branch, instead of the latest
release according to GitHub releases [1]. The latter might point at a
release built from the v0.10 release branch, which would potentially
cause problems when running with Cilium 1.11 and later.

[1] https://github.com/cilium/cilium-cli/releases/latest

nbusseneau reviewed (commented):
Q: why are we downloading the latest version in these workflows, whereas in others we have pinned the CLI version?

qmonnet commented on Documentation/gettingstarted/cli-download.rst:
```diff
@@ -7,7 +7,8 @@ various features (e.g. clustermesh, Hubble).
     .. code-block:: shell-session
 
-      curl -L --remote-name-all https://github.com/cilium/cilium-cli/releases/latest/download/cilium-linux-amd64.tar.gz{,.sha256sum}
+      export CILIUM_CLI_VERSION=$(curl -s https://raw.githubusercontent.com/cilium/cilium-cli/master/stable.txt)
```
Do we need the `export`s?

tklauser replied:
I don't think we do, will change.

qmonnet reviewed (approved):
Looks good to me, thanks!

tklauser committed:
docs, test/l4lb: use latest cilium-cli release according to stable.txt

This will make sure users download the latest release marked as stable
by developers in stable.txt on the master branch, instead of the latest
release according to GitHub releases [1]. The latter might point at a
release built from the v0.10 release branch, which would potentially
cause problems when running with Cilium 1.11 and later.

[1] https://github.com/cilium/cilium-cli/releases/latest

Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
ci: use pinned Cilium CLI version for post-test information gathering

Use the pinned Cilium CLI version instead of the latest GitHub release
which might point to a release cut from a branch which is not working
together with the Cilium version being tested.

Suggested-by: Nicolas Busseneau <nicolas@isovalent.com>
Signed-off-by: Tobias Klauser <tobias@cilium.io>
